cunumeric/module.py

```diff
@@ -1906,6 +1906,107 @@ def tile(A, reps):
     return result
 
 
+def repeat(a, repeats, axis=None):
+    """
+    Repeat elements of an array.
+
+    Parameters
+    ----------
+    a : array_like
+        Input array.
+    repeats : int or array of ints
+        The number of repetitions for each element. repeats is
+        broadcasted to fit the shape of the given axis.
+    axis : int, optional
+        The axis along which to repeat values. By default, use the
+        flattened input array, and return a flat output array.
+
+    Returns
+    -------
+    repeated_array : ndarray
+        Output array which has the same shape as a, except along the
+        given axis.
+
+    See Also
+    --------
+    numpy.repeat
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+
+    # when array is a scalar
+    if np.ndim(a) == 0:
+        if np.ndim(repeats) == 0:
+            return full((repeats,), a)
+        else:
+            raise ValueError(
+                "`repeat` with a scalar parameter `a` is only "
+                "implemented for scalar values of the parameter `repeats`."
+            )
+    if np.ndim(repeats) > 1:
+        raise ValueError("`repeats` should be scalar or 1D array")
+
+    # array is an array
+    array = convert_to_cunumeric_ndarray(a)
+    if np.ndim(repeats) == 1:
+        repeats = convert_to_cunumeric_ndarray(repeats)
+
+    # if no axes specified, flatten array
+    if axis is None:
+        array = array.ravel()
+        axis = 0
+
+    # axes should be integer type
+    if not isinstance(axis, int):
+        raise ValueError("Axis should be integer type")
+    axis = np.int32(axis)
+
+    if axis >= array.ndim:
+        return ValueError("axis exceeds dimension of the input array")
+
+    # If repeats is on a zero sized axis, then return the array.
+    if array.shape[axis] == 0:
+        return array.copy()
+
```

Comment:
Also handle the "broadcasting" case where `repeats.ndim == 1`, `repeats.shape[0] == 1 != array.shape[axis]`. In that case you can just get the only value in repeats and follow the scalar case:

```
>>> np.repeat([1,2,3], [2], axis=0)
array([1, 1, 2, 2, 3, 3])
```